### PR TITLE
youtube-dlc 2020.11.11-3 (new formula)

### PR DESCRIPTION
**Pull request by effleurager**
Adds a more complete formula for the community-fork youtube-dlc.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

In light of the takedown against and updates to `youtube-dl` since committed to `youtube-dlc`, this PR adds a formula for the new version. See the previous [PR](https://github.com/Homebrew/homebrew-core/pull/61125)'s [Reddit](https://www.reddit.com/r/DataHoarder/comments/ir8ic6/youtubedlc_an_active_fork_of_youtubedl/) post for a word from the maintainer of the original over 1-month old repo.

### CODE REVIEW

**Comment by robertleeblairjr** [comment]
Would this new formula be necessary? The youtube-dl maintainers have a gitlab repository and alternative site. Couldn't we just change the current youtube-dl formula to point to those sources which have been updated as recently as today? The current youtube-dl release is 2020.11.1.1.

**Comment by effleurager** [comment]
> The youtube-dl maintainers have a gitlab repository and alternative site.

This is a fork which has different maintainers and a more lax contribution process. It draws from their repo for changes more quickly than they merge, but it does stay current to their release branch.

> Couldn't we just change the current youtube-dl formula to point to those sources which have been updated as recently as today?

That sounds like it should be modified on the existing formula as a modification PR.

**Comment by robertleeblairjr** [comment]
> > The youtube-dl maintainers have a gitlab repository and alternative site.
> 
> This is a fork which has different maintainers and a more lax contribution process. It draws from their repo for changes more quickly than they merge, but it does stay current to their release branch.
> 
> > Couldn't we just change the current youtube-dl formula to point to those sources which have been updated as recently as today?
> 
> That sounds like it should be modified on the existing formula as a modification PR.

Gotcha. Thanks and I'll make a PR for those.

**Comment by robertleeblairjr** [comment]
> > The youtube-dl maintainers have a gitlab repository and alternative site.
> 
> This is a fork which has different maintainers and a more lax contribution process. It draws from their repo for changes more quickly than they merge, but it does stay current to their release branch.
> 
> > Couldn't we just change the current youtube-dl formula to point to those sources which have been updated as recently as today?
> 
> That sounds like it should be changed on the existing formula as a modification PR.

BTW. The GitHub repo this youtube-dlc references. Wouldn't it just be better to move the repos to other friendlier hosts? Obviously, GitHub is aggressively pursuing taking down any supported repos of youtube-dl and its forks. I would have thought removing any test files would satisfy the ban though.
Is this the new youtube-dl homepage?
[https://youtube-dl.org/](url)

File: Formula/youtube-dlc.rb
Comment (SMillerDev) [inline Formula/youtube-dlc.rb:15]: This isn't needed and should just be 
```suggestion
  head "https://github.com/blackjack4494/yt-dlc.git"
```

File: Formula/youtube-dlc.rb
Comment (SMillerDev) [inline Formula/youtube-dlc.rb:19]:
```suggestion
  depends_on "python@3.9"
```

File: Formula/youtube-dlc.rb
Comment (SMillerDev) [inline Formula/youtube-dlc.rb:11]: @samford @nandahkrishna  opinions?

Comment (nandahkrishna) [reply Formula/youtube-dlc.rb:11]: This is the default regex we use in case latest versions are tagged on GitHub, seems okay to me.

Comment (SMillerDev) [reply Formula/youtube-dlc.rb:11]: Does that work with date based versions?

Comment (nandahkrishna) [reply Formula/youtube-dlc.rb:13]: In this case, the date-based version is of the form `20xx.xx.xx`, which is matched by this regex. So, yes this should work.

Looking at the tags/previous releases, there's been a `2020.10.24-6` which wouldn't be matched by this regex. Modifying the regex to `%r{href=.*?/tag/v?(\d+(?:\.\d+)+(?:-\d+)?)["' >]}I` would match that version too, but I'd wait for @samford's opinion on this.

**Comment by MikeMcQuaid** [comment]
The `brew audit` states:

> youtube-dlc:
>  * GitHub repository too new (<30 days old)

I would like to sit on this for 30 days, as a result, until we see how the `youtube-dl` saga unfolds.

File: Formula/youtube-dlc.rb
Comment (samford) [reply Formula/youtube-dlc.rb:13]: We may want to simply modify the delimiter between the numbers to allow for `.` or `-`, like `%r{href=.*?/tag/v?(\d+(?:[.-]\d+)+)["' >]}i`. This accounts for the `2020.10.24-6` format but would also match unexpected version formats like `2020.10.24.6`, `2020-10-24`, `2020-10-24-6`, etc.

**Comment by DesWurstes** [comment]
```bash
$curl -H "Accept: application/vnd.github.v3+json" "https://api.github.com/repos/blackjack4494/yt-dlc"
  ...
  "created_at": "2020-10-24T04:23:41Z",
```

Today is its 31st day. Time for a version bump?


url "https://github.com/blackjack4494/yt-dlc/archive/2020.11.11-3.tar.gz"
sha256 "649f8ba9a6916ca45db0b81fbcec3485e79895cec0f29fd25ec33520ffffca84"

**Comment by effleurager** [comment]
All checks are now passing, and the repository is sufficiently old for consideration. 
Also, youtube-dl has had its repository visibility reinstated @MikeMcQuaid.

**Comment by MikeMcQuaid** [comment]
Given CI is passing I have no issue with this being merged as long as this fork remains active.

**Comment by DesWurstes** [comment]
> Given CI is passing I have no issue with this being merged as long as this fork remains active.

@MikeMcQuaid which means that the "do not merge" tag is going away?

File: Formula/youtube-dlc.rb
Comment (samford) [inline Formula/youtube-dlc.rb:8]:
```suggestion
  license "Unlicense"
  head "https://github.com/blackjack4494/yt-dlc.git"
```

There usually isn't a blank line before `head` when it's not a block, so we should remove it here.

File: Formula/youtube-dlc.rb
Comment (samford) [inline Formula/youtube-dlc.rb:15]: This may not be a necessary dependency but it would be good for someone else to give their thoughts on this.

We have nearly 2,000 formulae (at a minimum) that call `system "make"` but only six include `depends_on "make"`. This formula builds (and tests) fine without this `depends_on` line, so it probably makes sense to remove it.

**Review by samford — COMMENTED** [review]
The `livecheck` block looks good to me. I'll leave the final review/approval to someone more experienced with vetting new formulae but I've left comments about the areas that stuck out to me.

Once any changes have been addressed, it would be good to squash this branch into one "youtube-dlc 2020.11.11-3 (new formula)" commit.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@effleurager , thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**
